File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        if (n == 1) return 1;
        if (trust.length == 0) return -1;
        
        Set<Integer> trustiee = new HashSet<>();
        Set<Integer> person = new HashSet<>();
        for (int[] t : trust) {
            trustiee.add(t[0]);
            person.add(t[1]);
        }

        for (int p : person) {
            if (!trustiee.contains(p)) {
                int count = 0;
                //The town judge trusts nobody.
                for (int[] t : trust) {
                    if (t[1] == p) count++;
                }
                if (count == n - 1) return p;
            }
        }
        return -1;
//          Set<Integer> trustiee = new HashSet<>();
//          Set<Integer> person = new HashSet<>();
//          for (int[] t : trust) {
//              trustiee.add(t[0]);
//              person.add(t[1]);
//          }

//         int judge = -1;
//         for (int p : person) {
//             if (!trustiee.contains(p)) {
//                 judge = p;
//                 break;
//             }
//         }

//         return judge;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        get_trust = [0] * (n + 1)
        give_trust = [0] * (n + 1)

        for a, b in trust:
            give_trust[a] += 1
            get_trust[b] += 1

        for i in range(1, n + 1):
            if give_trust[i] == 0 and get_trust[i] == n - 1:
                return i

        return -1